pkg/controller/clusterprovision/clusterprovision_controller.go

```diff
@@ -303,9 +325,39 @@ func (r *ReconcileClusterProvision) reconcileRunningJob(instance *hivev1.Cluster
 		}
 	}
 
+	if time.Since(job.CreationTimestamp.Time) < podStatusCheckDelay {
+		timeUntilNextPodStatusCheck := job.CreationTimestamp.Time.Add(podStatusCheckDelay)
+		return reconcile.Result{RequeueAfter: time.Until(timeUntilNextPodStatusCheck)}, nil
+	}
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileClusterProvision) getInstallPod(job *batchv1.Job, pLog log.FieldLogger) (*corev1.Pod, error) {
+	podList := &corev1.PodList{}
+	podLabelSelector, err := metav1.LabelSelectorAsSelector(job.Spec.Selector)
+	if err != nil {
+		pLog.WithError(err).Error("could not create pod selector from job")
+		return nil, fmt.Errorf("could not create pod selector from job")
+	}
+	listOptions := &client.ListOptions{LabelSelector: podLabelSelector}
+	if err := r.List(context.TODO(), podList, listOptions); err != nil {
+		pLog.WithError(err).Error("could not list install pods")
+		return nil, fmt.Errorf("could not list install pods")
+	}
+
+	switch numberOfPods := len(podList.Items); {
+	case numberOfPods == 0:
+		pLog.Error("install pod not found")
+		return nil, fmt.Errorf("install pod not found")
+	case numberOfPods > 1:
+		pLog.Error("more than one install pod exists")
+		return nil, fmt.Errorf("more than one install pod exists")
+	}
+
+	installPod := &podList.Items[0]
+	return installPod, nil
```

Comment:
Take it or leave it, but I find this to be a bit cleaner.
```suggestion
	switch len(podList.Items) {
	case 0:
		pLog.Error("install pod not found")
		return nil, fmt.Errorf("install pod not found")
	case 1:
		return &podList.Items[0], nil
	default:
		pLog.Error("more than one install pod exists")
		return nil, fmt.Errorf("more than one install pod exists")
	}
```